lib/migration-stream.js

```diff
@@ -28,21 +29,44 @@ function MigrationStream() {
         return layer.version === 2;
     });
 
+    var newValidatorErrors; 
+
     if (v2 === true) {
-      // Check for errors on the v2 tile
-      var err = checkForErrors(info);
-      if (err) return callback(err);
 
-      migrationStream.push(tile);
-      return callback();
-    } else {
+      vtvalidate.isValid(tile.buffer, function(err, result){
+        // returns empty string if it's a valid tile
+        // if invalid tile - string that specifies why the tile is invalid
+        if (result){
+          newValidatorErrors = result; 
+        };
+
+        // Check for errors on the v2 tile
+        var data = {};
+        var err = checkForErrors(info);
```

Comment:
Noticing possible variable name collision with this `err` and the `err` returned from the `isValid()` callback.

Reply:
good catch, thanks!